### DISCUSSION TITLE
mdx type checking with Monaco editor

### DISCUSSION THREAD

**Discussion by websiddu** [mdx-js/mdx#2545] [Q&A]
I'm trying to figure out how to get type checking working for MDX files in Monaco Editor.

Let's say I have an MDX string and a type.d.ts file with type definitions. What are the core components and steps involved in getting Monaco to actually check the types in my MDX and show me errors?

Any insights or examples would be greatly appreciated!

**Comment by ChristianMurphy** [comment]
Welcome @websiddu! 👋 
Have you read the discussion at https://github.com/mdx-js/mdx-analyzer/issues/261
in particular the latest comment? https://github.com/mdx-js/mdx-analyzer/issues/261#issuecomment-1970626830

**Reply by websiddu** [reply]
Hmm. yes, I looked into it all I have implemented it got it working and some stuff works e.g. un-closed tags etc... however, I was not able to find ways to do type checking, and pass type file as, its monaco and I'm editing one file a time, it won't have access tsconfig like in a regular vs code editor. 

I would really appreciate if there were an example implementation of even pointing to the logic that is handing this type checking logic.

**Reply by remcohaszing** [reply]
I suggest you start by integrating `@volar/monaco` and make TypeScript support working. When that works, add `@mdx-js/language-service`.
